app/views/api/v2/job_invocations/main.json.rabl

```diff
@@ -21,6 +21,14 @@ child :targeting do
 
   child :hosts do
     extends 'api/v2/hosts/base'
+
+    if params[:host_status]
+      node(:job_status) do |host|
+        template_invocation = @job_invocation.template_invocations.find { |template_inv| template_inv.host_id == host.id }
+        task = template_invocation.try(:run_host_job_task)
```

Comment:
Are we checking permissions in here?

Reply:
The question is: Shouldn't we check permissions on `host` object, instead of `template_invocations`? Because we don't do that right now, see this scenario:
* Non-admin user with `view_hosts` filter, with `search` set to `aaa.example.com`
* Job invocation with three hosts: `aaa.example.com`, `bbb.example.com` & `ccc.example.com`
* Call `/api/job_invocations/934?host_status=true`

You should get results only for `aaa.example.com`, but you will get results for all three hosts.
In GUI works as expected, so the bug is only in API action.

[RM issue for host permission](https://projects.theforeman.org/issues/29972)